native/ortex/src/model.rs

```diff
@@ -83,27 +84,46 @@ pub fn run(
     model: ResourceArc<OrtexModel>,
     inputs: Vec<ResourceArc<OrtexTensor>>,
 ) -> Result<Vec<(ResourceArc<OrtexTensor>, Vec<usize>, Atom, usize)>, Error> {
-    // TODO: can we handle an error more elegantly than just .unwrap()?
+    // Grab the session and run a forward pass with it
+    let session: &ort::Session = &model.session;
 
     let mut ortified_inputs: Vec<ort::SessionInputValue> = Vec::new();
-    for input in inputs {
-        let derefed_input: &OrtexTensor = &input;
-        let v: ort::SessionInputValue = derefed_input.try_into()?;
-        ortified_inputs.push(v);
-    }
 
-    // Grab the session and run a forward pass with it
-    let session: &ort::Session = &model.session;
+    for (elixir_input, onnx_input) in zip(inputs, &session.inputs) {
+        let derefed_input: &OrtexTensor = &elixir_input;
+        if is_bool_input(&onnx_input.input_type) {
+            // this assumes that the boolean input isn't huge -- we're cloning it twice;
```

Comment:
Is there any danger in trying to cast a non-boolean input to boolean here? Seems it could succeed but not have the intended result?

Reply:
I don't think there's danger in terms of correctness. But it does allow users to cast non-bool Nx tensors (i.e., u8s) into onnx bool-tensors, which might be counter-intuitive.

I like the idea of being consistent with Nx typing, and panicking if we try to bool-ify a non u8 input. and I'll modify the OrtexTensor to_bool() function (which should also simplify some of the code!)